build.rs

```diff
@@ -41,71 +45,199 @@ fn wast_processor(out: &mut Testsuite, p: PathBuf) -> Option<Test> {
     })
 }
 
+/// Given a Testsuite and a path, process the path in case is a Emscripten
+/// wasm file.
+fn emscripten_processor(out: &mut Testsuite, p: PathBuf) -> Option<Test> {
```

Comment:
I'm still not a fan of including specific logic in a workspace build.rs like this:

It's a relatively minor point and as our build.rs is currently only doing 1 thing the only cost is increasing the amount of visible complexity in the codebase.  And my real concern is about setting a bad example for future editors: ideally the correct thing (keeping test generation and other compile-time logic separate) should be easy, otherwise people won't do it.

So not the biggest deal, but I do think this does increase the chance of having a tangled mess that we'll have to untangle in the future (it's not at all right now, but after N more features get added it might be).

Reply:
Yeah, I will move the processors back to the tests.
The general body of `main` in build.rs I think it's already minimal, but descriptive, so I think we can take out almost everything out of main.

Reply:
Perhaps there are other ways of doing this, let me know if you have ideas @MarkMcCaskey!

Reply:
Fixed!